main.tf

```diff
@@ -36,12 +31,12 @@ resource "azurerm_network_security_group" "nsg" {
 }
 
 ##-----------------------------------------------------------------------------
-## Below resource will create network security group inbound rules in azure and will be attached to above network security group.  
+## Below resource will create network security group inbound rules in azure and will be attached to above network security group.
 ##-----------------------------------------------------------------------------
 resource "azurerm_network_security_rule" "inbound" {
   for_each                     = { for rule in var.inbound_rules : rule.name => rule }
   resource_group_name          = var.resource_group_name
-  network_security_group_name  = join("", azurerm_network_security_group.nsg.*.name)
+  network_security_group_name  = join("", azurerm_network_security_group.nsg[*].name)
```

Comment:
Pls remove join and use [0] instead of [*]

Reply:
fixed